entities/TV.gd

```diff
@@ -0,0 +1,35 @@
+"""
+Logic of the TV
+"""
+extends "res://sources/entity/Interactable.gd"
+
+export (PackedScene) var puzzle_scene
+
+
+func _ready() -> void:
+	assert(puzzle_scene)
+
+
+func on_interacted(_actor: Node2D) -> void:
+	var puzzle = Utils.get_game().puzzle_start(puzzle_scene)
```

Comment:
I think TV don't need to control state of puzzle. It should emit a signal only. Level manager will handle the signal, check if it solved or not and decide if puzzle can start.

Reply:
Yeah you are right, it will make the general behavior of a level easier to grasp if we do that in the level script. I'll update that